drag-and-drop.md

```diff
@@ -0,0 +1,126 @@
+# Drag-adn-Drop
+
+The proposal to provide native support Drag-and Drop opeerations:
+* drag and drop between widgets 
+* receive drag and drop URL(s) to application from a desktop or another application
+* start drag operation from Fyne application with URL(s) to another desltop application (pending external changes)
+
+## Background
+
+There are few issues opened and related to missing drag and drop functionality:
+* Add drag and drop support https://github.com/fyne-io/fyne/issues/142
+* Drag and drop to application icon on macOS https://github.com/fyne-io/fyne/issues/497
+
+Existing interface provide a minimum capabilities to programmatically start a drug operation and get notified when it complets.
+This functionality doesn't provide an interface to communicate between widgets within the application nor send/receive drag and drop with other applications.
+
+## Architecture / API
+
+Existing interface will not change to preserve compatibility with existing code.
+The proposed changes will extend the interface and give user missing capabilities.
+
+The **fyne.CanvasObject** wishing to initiate drag operation must implement **fyne.Draggable** interface. No interface changes.
+
+The **fyne.CanvasObject** wishing to receiver the drop must implement **fyne.DragReceiver** interface.
+
+```
+package fyne
+
+type DragReceiver interface {
+	DragDrop(Draggable)
+}
+```
+
+To pass information to receiver the **fyne.Draggable** can also implement **drag.Info** interface.
+
+```
+package drag
+
+type Info interface {
+	Payload() Payload
+}
+```
+
+To initiate a drag-and-drop outside the Fyne application the object implement **drag.AppDraggable** interface. This interface will allow the object prepare the **drag.Payload**.
```

Comment:
This should probably be more specific about how it relates to starting a drop from an app icon etc. The text implies it is for window drops which it isn't.

Reply:
I think adding drag-and-drop of a CanvasObject to another desktop application is confusing this design proposal.
I think this portion of design better removed for now sine it has an implementation constraint with go-gl/glfw.
I will move this to the end of the document with a note.

Reply:
Quite possibly - that would mean that we can't use the `fyne.Draggable` interface any more I guess